packages/models/src/models/Subscriptions.ts

```diff
@@ -2045,4 +2073,20 @@ export class SubscriptionsRaw extends BaseRaw<ISubscription> implements ISubscri
 
 		return this.updateOne(query, update);
 	}
+
+	async updateOneAndSyncRolePriority(query: Filter<ISubscription>, update: UpdateFilter<ISubscription>): Promise<UpdateResult> {
+		const updateResult = await this.updateOne(query, update);
+
+		const subscription = await this.findOne(query, { projection: { roles: 1, u: 1, rid: 1 } });
+
+		if (!subscription) {
+			return updateResult;
+		}
+
+		const rolePriority = subscription.roles ? calculateRoomRolePriorityFromRoles(subscription.roles) : ROOM_ROLE_PRIORITY_MAP.default;
```

Comment:
Ideally we shouldn't have dependencies between models, i think most of them were removed already. Can you, instead of this, just call the `addRoomRolePriority` where the other functions are called, so the call for users ends outside of the model?